### DISCUSSION TITLE
Real support for Information requests

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5703] [Feature Requests]
> Information Requests
> 
> > Collection managers are regularly asked to share specific data records with local or external researchers. In most cases this is accomplished by attaching a spreadsheet or .csv file to an email message. The information shared is usually in the form of Collection Objects. Specify 6 has an automated function to handle responses to specimen information requests and record these for later retrieval and reporting.
> > 
> > Interactions such as Loans, Gifts, Exchange Outs and Disposals can also be created using Information Requests.
> > 
> > Click _**Information Request**_ in the side bar to start the process.
> > 
> > If any record sets have been saved, a dialog appears prompting the user to either Enter Catalog Numbers or use an existing record set.  The Record Set method would be easiest when using large sets of data.
> 
> *   **Enter Catalog Numbers** by typing catalog numbers into the text dialog, separating multiple numbers with commas.
>     
> *   Select a **Record** Set from the list.  If only one record set exists, it will be used by default.  This can also be accomplished by dragging and dropping the record set onto the Information Request item in the sidebar.
> 
> > The Collection Objects will be added to the Information Request form.  Once they are added, the _Send Email on Save_ check box becomes available on the form.  If checked and a) the recipient has an email address associated with their agent name in the database and b) the necessary email settings have been entered in _Preferences > Email_, an email will automatically be sent to the designated recipient once the Save button has been clicked.  The attached Collection Objects will appear in the main portion of the email and will also be attached as a CSV file.
> > 
> > Unlike queries, Information Requests are only saved on the side bar and not in the database and are not searchable.  Saved Information Requests listed on the side bar can be deleted by dragging and dropping them onto the trash can.  No record will be kept in the database of an Information Request once it has been deleted.
> > 
> > It is possible to send an Information Request via email at a later date or resend it if it has been saved.
> 
> *   Click on the _Information Request_ button in the side bar to open it in the workspace.
> *   Click the (edit form) button at the bottom of the workspace to edit the Information Request.
> *   Make a change to the Information Request (this could simply be retyping part of the date).
> *   Check the _Send Email on Save_ button.
> *   Click the _Save_ button
>

**Comment by grantfitzsimmons** [comment]
**Reported by:** Garth at UMich

**Comment by beach53** [comment]
I wonder about the importance of this, sending email from within Specify raises all sorts of problems.  They can just export a file make it an attachment and send it to them in their own email app.  I am happy to discuss but this always seemed like a frill.

**Comment by maxpatiiuk** [comment]
we can have a link, than when clicked, opens a new draft email in the user system's default email client (which could be gmail) with subject and body pre-composed by specify - and then the user sends the email
so specify isn't responsible for the email server, but does automate part of the process

implementing that doesn't look complicated assuming the browser support is good - https://www.computerhope.com/issues/ch000056.htm

**Comment by bronwyncombs** [comment]
requested by ally coconis @ Santa Barbara Museum of Natural History via [forum](https://discourse.specifysoftware.org/t/information-request-issue-adding-specimen-data/3821?u=bronwyn): 

> In Specify 7, I am trying to create an information request transaction. However, when I add a record set to the form, I cannot see the objects attached to the record set. The form creates a single row of blank boxes that it then requires me to fill in such as catalog number, field number, and locality, amongst other fields. Can you explain this to me? It was my understanding that I would create a record set and just attach this to the information request. Thank you for the help.
